packages/kbn-telemetry-tools/src/tools/manage_schema.ts

```diff
@@ -28,7 +28,7 @@ export type AllowedSchemaTypes =
   | 'date'
   | 'float';
 
-export function compatibleSchemaTypes(type: AllowedSchemaTypes) {
+export function compatibleSchemaTypes(type: AllowedSchemaTypes | 'array') {
```

Comment:
Why didnt we add `array` to the `AllowedSchemaTypes` type here instead of using union?

Reply:
Because I think it would allow `{ type: 'array' }` without items. Wouldn't it?